crates/apub/src/activities/voting/vote.rs

```diff
@@ -0,0 +1,133 @@
+use crate::{
+  activities::{
+    community::announce::AnnouncableActivities,
+    generate_activity_id,
+    verify_activity,
+    verify_person_in_community,
+    voting::{vote_comment, vote_post},
+  },
+  activity_queue::send_to_community_new,
+  extensions::context::lemmy_context,
+  fetcher::{
+    objects::get_or_fetch_and_insert_post_or_comment,
+    person::get_or_fetch_and_upsert_person,
+  },
+  ActorType,
+  PostOrComment,
+};
+use anyhow::anyhow;
+use lemmy_api_common::blocking;
+use lemmy_apub_lib::{values::PublicUrl, ActivityCommonFields, ActivityHandler};
+use lemmy_db_queries::Crud;
+use lemmy_db_schema::{
+  source::{community::Community, person::Person},
+  CommunityId,
+};
+use lemmy_utils::LemmyError;
+use lemmy_websocket::LemmyContext;
+use serde::{Deserialize, Serialize};
+use std::{convert::TryFrom, ops::Deref};
+use strum_macros::ToString;
+use url::Url;
+
+#[derive(Clone, Debug, ToString, Deserialize, Serialize)]
+pub enum VoteType {
+  Like,
+  Dislike,
+}
```

Comment:
I think this enum (or something similar) would also be very useful to deduplicate the database and api code. So we dont need the same logic twice, for upvote and for downvote.

Reply:
Currently in the API, its just `score`, which must be either -1, 0, or 1, for dislike, undo like, or like.

Reply:
Yes of course it must be serialized, but using an enum would make the code easier to understand.